lib/java/src/org/apache/thrift/TApplicationException.java

```diff
@@ -73,7 +74,18 @@ public int getType() {
     return type_;
   }
 
-  public static TApplicationException read(TProtocol iprot) throws TException {
```

Comment:
Technically this is a breaking change.  However this is not a method that would typically be used by applications I think, so the impact would probably be very low... and even for affected applications, it should be easy to fix (that is, create a new `TApplicationException` and call the read method).  It might be nice to add a static convenience method as a replacement (named `TApplicationException.readFrom(protocol)` or something like that)